nickytonline/astro-partykit-starter
# feat: set up Vitest

## Description

This PR adds [Vitest](https://vitest.dev/) to the project with details as follow:

- Install Vitest
- Add `vitest.config.ts` and the configuration.
- Add `test` to scripts in `package.json`.
- Add types for Vitest in `tsconfig.json`.
- Add a dummy test in `basics.tests.ts`.
- Update `lint-staged` with vitest.
- Add command to run Vitest to README.



## What type of PR is this? (check all applicable)


## Related Tickets & Documents

Closes #40 


## Mobile & Desktop Screenshots/Recordings




## Steps to QA

1. Run `npm install`.
2. Go to `test` folder inside `src` folder.
3. Open `basic.tests.ts`.
4. Run `npm run test`.
5. You will see on the terminal that 2 tests in 1 file are passed.
6. Change a number in the 1st test or a word in the 2nd test.
7. You will see the information of which line of code is not correct and that there is 1 failed test in 1 file.





## Added to documentation?


## [optional] Are there any post-deployment tasks we need to perform?



## [optional] What gif best describes this PR or how it makes you feel?

![](https://media.giphy.com/media/CS8Fgb5BAynaDGY1Me/giphy.gif)
